Rakefile

```diff
@@ -227,7 +227,7 @@ if defined? JRUBY_VERSION
       driver_jars
     end
 
-    if ENV['RUBYARCHDIR'] || ENV['BUILD_EXT_MAVEN'] == 'true'
+    if ENV['BUILD_EXT_MAVEN'] == 'true'
```

Comment:
at this point I am not sure why I did it (maybe its to be found in support branches' history) 
but you need to carefully evaluate all possibilities (some are in the comments) of the ext building being triggered.
e.g. with/without Bundler

Reply:
> There was a check on RUBYARCHDIR in the Rake task for the adapter jar which used to be set by the build script. This is not set anymore, so I removed the check.

we certainly weren't setting RUBYARCHDIR - believe it was simply from how the ENV ends up being when the ext is going to get build